File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/domain/medal/UserMedalCreator.java
package com.depromeet.threedollar.domain.domain.medal;

import com.depromeet.threedollar.common.docs.ObjectMother;
import com.depromeet.threedollar.domain.domain.user.User;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class UserMedalCreator {

    public static UserMedal createActive(Medal medal, User user) {
        return new UserMedal(medal, user, UserMedalStatus.ACTIVE);
    }

    public static UserMedal createInActive(Medal medal, User user) {
        return new UserMedal(medal, user, UserMedalStatus.IN_ACTIVE);
    }

}

File: threedollar-domain/src/testFixtures/java/com/depromeet/threedollar/domain/user/domain/medal/UserMedalCreator.kt
package com.depromeet.threedollar.domain.user.domain.medal

import com.depromeet.threedollar.domain.user.domain.TestFixture
import com.depromeet.threedollar.domain.user.domain.user.User

@TestFixture
object UserMedalCreator {

    @JvmStatic
    fun createActive(
        medal: Medal,
        user: User
    ): UserMedal {
        return UserMedal.builder()
            .medal(medal)
            .user(user)
            .status(UserMedalStatus.ACTIVE)
            .build()
    }

    @JvmStatic
    fun createInActive(
        medal: Medal,
        user: User
    ): UserMedal {
        return UserMedal.builder()
            .medal(medal)
            .user(user)
            .status(UserMedalStatus.IN_ACTIVE)
            .build()
    }

}
